internal/controllers/bundle/bundle.go

```diff
@@ -159,19 +159,19 @@ func (c *controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	// a potential deletion.
 	if !equality.Semantic.DeepEqual(existingBundle.Status, reconciledBundle.Status) {
 		if updateErr := c.cl.Status().Update(ctx, reconciledBundle); updateErr != nil {
-			return res, apimacherrors.NewAggregate([]error{reconcileErr, updateErr})
+			return ctrl.Result{}, apimacherrors.NewAggregate([]error{reconcileErr, updateErr})
 		}
 	}
 	existingBundle.Status, reconciledBundle.Status = rukpakv1alpha1.BundleStatus{}, rukpakv1alpha1.BundleStatus{}
 	if !equality.Semantic.DeepEqual(existingBundle, reconciledBundle) {
 		if updateErr := c.cl.Update(ctx, reconciledBundle); updateErr != nil {
-			return res, apimacherrors.NewAggregate([]error{reconcileErr, updateErr})
+			return ctrl.Result{}, apimacherrors.NewAggregate([]error{reconcileErr, updateErr})
 		}
 	}
-	return res, reconcileErr
+	return ctrl.Result{}, reconcileErr
 }
 
-func (c *controller) reconcile(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (ctrl.Result, error) {
```

Comment:
Here and in `internal/controllers/bundledeployment/bundledeployment.go` we always return an empty value in `ctrl.Result{}`. It makes sense in `Reconcile` to satisfy the interface, but not in this unexported func.

Reply:
I'm not sure I agree with this change. If we ever _do_ want to return something other than `ctrl.Result{}`, it'll likely be coming from `reconcile()`, not `Reconcile()`.

I'd be slightly worried that our future selves would be less inclined to change the return values back to `(ctrl.Result, error)` and would instead do something less desirable overall.

The primary purpose of this split is to make us not really have to think at all about status updating. The result is merely passed through.

Is a linter complaining about this? If so, WDYT about applying the `//nolint:<whatever>` label with an explanation along these lines.

Reply:
@joelanford I also was not super happy about this linting issue at first, but ended up obeying because I convinced myself that the code is not set in stone. It is super easy to add a new return param once we have a need for it (especially since this is not an exported method).

To me it is the same as adding input arguments into a function: we usually add new args as the need for them arises. There are exceptions (e.g. designing a public interfaces where signature changes are more painful), but it is not the case here.

But I can look into either adding nolint or disabling this linting rule altogether tomorrow. I am not a big fan of inline linting exceptions, but also do not have very strong objections against them.